src/main/java/com/spotify/dns/LookupFactory.java

```diff
@@ -17,15 +17,28 @@
 package com.spotify.dns;
 
 import org.xbill.DNS.Lookup;
+import org.xbill.DNS.lookup.LookupSession;
 
 /**
- * Library-internal interface used for finding or creating {@link Lookup} instances.
+ * Library-internal interface used for finding or creating {@link LookupSession} or {@link Lookup} instances.
  */
 interface LookupFactory {
   /**
    * Returns a {@link Lookup} instance capable of doing SRV lookups for the supplied FQDN.
+   * @deprecated
+   * This synchronous method is being deprecated.
+   * Use {@link LookupFactory#sessionForName(String)} instead
+   *
    * @param fqdn the name to do lookups for
    * @return a Lookup instance
    */
+  @Deprecated
   Lookup forName(String fqdn);
```

Comment:
Is this method really used outside of the library? The interface is already package private, so consumers would call either `SimpleLookupFactory` or `CachingLookupFactory` directly. If that's not used anywhere, there's not much point in keeping it and all that depends on it...

Reply:
But we can't remove it either, as it's being used by the synchronous `resolve` method.